pkg/grpc/client.go

```diff
@@ -36,36 +35,29 @@ type Client struct {
 	host             string
 	timeoutSeconds   int
 	insecure         bool
-	grpcConnectOnce  *sync.Once
 	connClose        func() error
 	conn             *grpc.ClientConn
 	descriptorSource grpcurl.DescriptorSource
 }
 
 // NewClient returns a gRPC client.
 func NewClient(host string, insecure bool, timeoutSeconds int) Client {
-	return Client{host: host, timeoutSeconds: timeoutSeconds, grpcConnectOnce: new(sync.Once), insecure: insecure, connClose: func() error { return nil }}
+	return Client{host: host, timeoutSeconds: timeoutSeconds, insecure: insecure, connClose: func() error { return nil }}
 }
 
 // connect attempts to establish a connection with a gRPC server.
-func (c *Client) connect(headers []string) error {
-
-	dialTime := 10 * time.Second
```

Comment:
The gRPC client by its nature will try to reconnect with backoff whenever this is needed. We can simply use the `timeoutSeconds` without an additional timeout for establishing the connection.